platforms/core-configuration/declarative-dsl-core/src/main/kotlin/org/gradle/internal/declarativedsl/intrinsics/IntrinsicTopLevelFunctionBridge.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.declarativedsl.intrinsics
+
+import org.gradle.declarative.dsl.schema.DataTopLevelFunction
+import org.gradle.internal.declarativedsl.mappingToJvm.RuntimeFunctionCandidatesProvider
+import java.lang.reflect.Modifier
+import kotlin.reflect.KClass
+import kotlin.reflect.KFunction
+import kotlin.reflect.full.findAnnotation
+import kotlin.reflect.jvm.kotlinFunction
+
+/**
+ * Annotates a bridge function needed to invoke top-level functions that are otherwise non-callable via reflection, such as the top-level functions in kotlin-stdlib.
+ *
+ * For instance, `kotlin.collections.CollectionsKt` inherits its functions from non-public multi-file fragment classes like `kotlin.collections.CollectionsKt_CollectionsJvmKt`.
+ * The functions inherited that way cannot be called via reflection.
+ */
+@Retention(AnnotationRetention.RUNTIME)
+@Target(AnnotationTarget.FUNCTION)
+annotation class IntrinsicTopLevelFunctionBridge(val topLevelFunctionFqn: String)
```

Comment:
Not that I can come up with a better name, but this "bridge" thing feels confusing... You can understand it from the annotation's Javadoc, but not so from just the annotation's name. I'll try to suggest an improvement, but atm I'm shooting blanks...

Reply:
FWIW I'm fine with the current name

Reply:
I, too, was not able to identify a direction in which I could change this name to make it clearer. A "bridge" is a common term for trampoline declarations (although often synthetic) that get invoked in place of something else that can't be invoked directly, so I'd like to keep that word here. Does "intrinsic" sound wrong?